packages/optimizely-cms-sdk/src/render/assets.ts

```diff
@@ -0,0 +1,191 @@
+import type { InferredContentReference } from '../infer.js';
+import { appendToken } from '../util/preview.js';
+
+/**
+ * Appends a preview token to a ContentReference's rendition URLs.
+ * Creates a new object with modified renditions to avoid mutation.
+ * Internal function - not exported.
+ *
+ * @param input - ContentReference from a DAM asset
+ * @param previewToken - The preview token to append to rendition URLs
+ * @returns New ContentReference with preview tokens appended to rendition URLs
+ */
+function appendPreviewTokenToRenditions(
+  input: InferredContentReference | null | undefined,
+  previewToken: string | undefined
+): InferredContentReference | null | undefined {
+  if (!input || !previewToken) return input;
+
+  // Create a shallow copy of the input
+  const result = { ...input };
+
+  // Append preview token to all rendition URLs if they exist
+  if (result.item && 'Renditions' in result.item && result.item.Renditions) {
+    result.item = {
+      ...result.item,
+      Renditions: result.item.Renditions.map((r) => ({
+        ...r,
+        Url: r.Url ? appendToken(r.Url, previewToken) : r.Url,
+      })),
+    };
+  }
+
+  return result;
+}
+
+/**
+ * Creates a responsive srcset string from your image renditions.
+ *
+ * This handles all the messy details:
+ * - Removes duplicate widths if you have multiple renditions at the same size
+ * - Adds preview tokens automatically when in edit mode
+ * - Returns an empty string if there's no image or no renditions
+ *
+ * @param opti - Your content object with __context for preview tokens
+ * @param property - The image reference from your content (e.g., opti.image)
+ * @returns A srcset string like "url1 100w, url2 500w" or empty string
+ *
+ * @example
+ * ```tsx
+ * import { damAssets } from '@optimizely/cms-sdk';
+ *
+ * export default function MyComponent({ opti }) {
+ *   const { getSrcset } = damAssets(opti);
+ *
+ *   return (
+ *     <img
+ *       src={opti.image?.item?.Url}
+ *       srcSet={getSrcset(opti.image)}
+ *       sizes="(max-width: 768px) 100vw, 50vw"
+ *     />
+ *   );
+ * }
+ * ```
+ *
+ * @example
+ * Works with any image property:
+ * ```tsx
+ * const { getSrcset, getAlt } = damAssets(opti);
+ * <img srcSet={getSrcset(opti.heroImage)} alt="Hero" />
+ * ```
+ */
+export function getSrcset<T extends Record<string, any>>(
+  opti: T & { __context?: { preview_token?: string } },
+  property: InferredContentReference | null | undefined
```

Comment:
Is this correct?

If a developer passes null or undefined, this will return empty string resulting in:

```
<img srcset="" />
```

Is it correct?